diff --git a/src/balances.rs b/src/balances.rs
index 282ea93..e26f5ec 100644
--- a/src/balances.rs
+++ b/src/balances.rs
@@ -1,16 +1,16 @@
 use std::collections::BTreeMap;
 
 /// This is the Balances Module.
 /// It is a simple module which keeps track of how much balance each account has in this state
 /// machine.
 pub struct Pallet {
 	// A simple storage mapping from accounts (`&'static str`) to their balances (`u128`).
-	/* TODO: Add a field `balances` which is a `BTreeMap` fom `&'static str` to `u128`. */
+	balances: BTreeMap<&'static str, u128>,
 }
 
 impl Pallet {
 	/// Create a new instance of the balances module.
 	pub fn new() -> Self {
-		/* TODO: Return a new instance of the `Pallet` struct. */
+		Self { balances: BTreeMap::new() }
 	}
 }
