diff --git a/src/balances.rs b/src/balances.rs
index e26f5ec..9f4fcd7 100644
--- a/src/balances.rs
+++ b/src/balances.rs
@@ -13,4 +13,15 @@ impl Pallet {
 	pub fn new() -> Self {
 		Self { balances: BTreeMap::new() }
 	}
+
+	/// Set the balance of an account `who` to some `amount`.
+	pub fn set_balance(&mut self, who: &'static str, amount: u128) {
+		/* Insert `amount` into the BTreeMap under `who`. */
+	}
+
+	/// Get the balance of an account `who`.
+	/// If the account has no stored balance, we return zero.
+	pub fn balance(&self, who: &'static str) -> u128 {
+		/* Return the balance of `who`, returning zero if `None`. */
+	}
 }
